rskj-core/src/main/java/co/rsk/peg/federation/PendingFederation.java

```diff
@@ -97,47 +98,42 @@ public PendingFederation addMember(FederationMember member) {
      */
     public Federation buildFederation(
-        Instant creationTime,
-        long blockNumber,
-        BridgeConstants bridgeConstants,
-        ActivationConfig.ForBlock activations
+            Instant creationTime,
+            long blockNumber,
+            BridgeConstants bridgeConstants,
+            ActivationConfig.ForBlock activations
         ) {
         if (!this.isComplete()) {
             throw new IllegalStateException("PendingFederation is incomplete");
         }
 
-        if (activations.isActive(ConsensusRule.RSKIP353)) {
-            logger.info("[buildFederation] Going to create a P2SH ERP Federation");
-            return FederationFactory.buildP2shErpFederation(
-                members,
-                creationTime,
-                blockNumber,
-                bridgeConstants.getBtcParams(),
-                bridgeConstants.getErpFedPubKeysList(),
-                bridgeConstants.getErpFedActivationDelay()
-            );
+        NetworkParameters btcParams = bridgeConstants.getBtcParams();
+        FederationArgs federationArgs = new FederationArgs(members, creationTime, blockNumber, btcParams);
+
+        if (shouldBuildStandardMultisigFederation(activations)){
+            return FederationFactory.buildStandardMultiSigFederation(federationArgs);
         }
 
-        if (activations.isActive(ConsensusRule.RSKIP201)) {
-            logger.info("[buildFederation] Going to create an ERP Federation");
+        // should build and erp federation due to activations
+        List<BtcECKey> erpPubKeys = bridgeConstants.getErpFedPubKeysList();
+        long activationDelay = bridgeConstants.getErpFedActivationDelay();
+        ErpFederationArgs erpFederationArgs = ErpFederationArgs.fromFederationArgs(federationArgs, erpPubKeys, activationDelay);
 
-            return FederationFactory.buildNonStandardErpFederation(
-                members,
-                creationTime,
-                blockNumber,
-                bridgeConstants.getBtcParams(),
-                bridgeConstants.getErpFedPubKeysList(),
-                bridgeConstants.getErpFedActivationDelay(),
-                activations
-            );
+        if (shouldBuildNonStandardErpFederation(activations)) {
+            logger.info("[buildFederation] Going to create an ERP Federation");
+            return FederationFactory.buildNonStandardErpFederation(erpFederationArgs, activations);
         }
 
-        return FederationFactory.buildStandardMultiSigFederation(
-                members,
-                creationTime,
-                blockNumber,
-                bridgeConstants.getBtcParams()
-        );
+        logger.info("[buildFederation] Going to create a P2SH ERP Federation");
+        return FederationFactory.buildP2shErpFederation(erpFederationArgs);
+    }
+
+    private boolean shouldBuildStandardMultisigFederation(ActivationConfig.ForBlock activations) {
+        return !activations.isActive(ConsensusRule.RSKIP201);
```

Comment:
```suggestion
        return !activations.isActive(ConsensusRule.RSKIP201) && !activations.isActive(ConsensusRule.RSKIP353);
```
not sure it's worth it, what do you think?

Reply:
mmm... it is assumed that if RSKIP201 is not active, RSKIP353 is less so. If we end up having that situation we would probably be facing a bigger problem